src/box/lua/load_cfg.lua

```diff
@@ -1149,7 +1149,24 @@ local function get_option_from_env(option)
 
     -- This code lean on the existing set of template_cfg
     -- types for simplicity.
-    if param_type:find('table') and raw_value:find(',') then
+    if param_type:find('table') and (raw_value:startswith('{') or
+                                     raw_value:startswith('[')) then
+        return json.decode(raw_value)
+    elseif param_type:find('table') and raw_value:find('=') then
+        assert(not param_type:find('boolean'))
+        local res = {}
+        for _, v in ipairs(raw_value:split(',')) do
+            local eq = v:find('=')
+            if eq == nil then
+                error(err_msg_fmt:format(env_var_name, option,
+                                         'in `key=value` or `value` format'))
+            end
+            local lhs = string.sub(v, 1, eq - 1)
```

Comment:
It seems that there must be an error if `lhs` is an empty string. And there must be a test case that checks it.

Reply:
Done.